stl/inc/span

```diff
@@ -0,0 +1,665 @@
+        // non-member overload to model the exposition-only forwarding-range concept
+        return _Right.begin();
+    }
+    _NODISCARD friend constexpr const_iterator end(span _Right) noexcept {
+        // non-member overload to model the exposition-only forwarding-range concept
+        return _Right.end();
+    }
+#endif // __cpp_lib_concepts
+
+    _NODISCARD constexpr const_iterator cbegin() const noexcept {
+        return begin();
+    }
+
+    _NODISCARD constexpr const_iterator cend() const noexcept {
+        return end();
+    }
+
+    _NODISCARD constexpr reverse_iterator rbegin() noexcept {
+        return reverse_iterator{end()};
+    }
+
+    _NODISCARD constexpr reverse_iterator rend() noexcept {
+        return reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rbegin() const noexcept {
+        return const_reverse_iterator{end()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rend() const noexcept {
+        return const_reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crbegin() const noexcept {
+        return rbegin();
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crend() const noexcept {
+        return rend();
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_begin() noexcept {
+        return _Mydata;
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_begin() const noexcept {
+        return _Mydata;
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_end() noexcept {
+        return _Mydata + size();
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_end() const noexcept {
+        return _Mydata + size();
+    }
+
+private:
+    pointer _Mydata{nullptr};
+};
+
+// [span.tuple] Tuple interface
+template <class _Ty, size_t _Extent>
+struct tuple_size<span<_Ty, _Extent>> : integral_constant<size_t, _Extent> {
+    static_assert(dynamic_extent != _Extent, "std::span<T, dynamic_extent> is not tuple-like. ");
+};
+
+template <size_t _Index, class _Ty, size_t _Extent>
+struct tuple_element<_Index, span<_Ty, _Extent>> {
+    static_assert(dynamic_extent != _Extent, "std::span<T, dynamic_extent> is not tuple-like. ");
+    static_assert(_Index < _Extent, "Index out of bounds for a std::span of this extent. ");
+    using type = _Ty;
+};
+
+template <size_t _Index, class _Ty, size_t _Extent>
+_NODISCARD constexpr _Ty& get(span<_Ty, _Extent> _Span) noexcept {
+    static_assert(dynamic_extent != _Extent, "std::get<> not supported for std::span<T, dynamic_extent>");
+    static_assert(_Index < _Extent, "Index out of bounds for a std::span of this extent. ");
+    return _Span[_Index];
+}
+
+#if _HAS_STD_BYTE
+// [span.objectrep] Views of object representation
+// FUNCTION TEMPLATE as_bytes
+template <class _Ty, size_t _Extent, enable_if_t<_Extent <= dynamic_extent / sizeof(_Ty), int> = 0>
+_NODISCARD auto as_bytes(span<_Ty, _Extent> _Sp) noexcept -> span<const byte, sizeof(_Ty) * _Extent> {

```

Comment:
Nitpick: This is inconsistent about naming single arguments as `_Span` (in `get`) or `_Sp` (in `as_bytes` etc.). In general, the same name should be used unless the contexts are significantly different (e.g. `_Other` in a converting constructor is good because it's the "other" span). I like `_Span` just fine for these functions.

Reply:
I went with `_Span`